### PR TITLE
Replace fhirstore by hapi

### PR DESCRIPTION
**Pull request by simonvadee**
Fixes E2E tests.
removed fhirstore dependency. Added dependency to https://github.com/beda-software/fhir-py. I wanted to use https://github.com/smart-on-fhir/client-py but there is no way to pass a custom auth header...

### CODE REVIEW

File: tests/river/test_cancel_batch.py [hunk 1]
```diff
@@ -37,10 +26,12 @@ def cancel_batch(batch_id):
     except requests.exceptions.ConnectionError:
         raise Exception("Could not connect to the api service")
 
-    assert response.status_code == 200, f"api DELETE /batch/{batch_id}/ returned an error: {response.text}"
+    assert (
+        response.status_code == 200
```
Comment (tevariou) [inline tests/river/test_cancel_batch.py:30]: irrelevant here but maybe we should change that to be a 204

File: docker-compose.yml [hunk 2]
```diff
@@ -9,12 +9,9 @@ services:
     build: .
     restart: on-failure
     environment:
+      - FHIR_API_URL=${FHIR_API_URL:-jpaltime:8080/hapi/fhir}
+      - FHIR_API_AUTH_TOKEN={FHIR_API_AUTH_TOKEN:-none}
```
Comment (tevariou) [inline docker-compose.yml:13]: None? or just `FHIR_API_AUTH_TOKEN={FHIR_API_AUTH_TOKEN}` ?

**Review by tevariou — APPROVED** [review]

File: tests/river/conftest.py [hunk 3]
```diff
@@ -62,10 +56,19 @@ def _credentials_factory(source_id: str, credentials: dict):
 
 
 @pytest.fixture(scope="session")
-def concept_maps_factory(fhirstore: FHIRStore):
+def concept_maps_factory(fhir_client: SyncFHIRClient):
     @contextlib.contextmanager
```
Comment (Jasopaum) [inline tests/river/conftest.py:60]: Why is this a context?

File: tests/river/test_batch.py [hunk 4]
```diff
@@ -101,19 +105,24 @@ def test_batch(pyrog_resources, batch):
...
-# TODO: check in elastic that references have been set
+    result = (
+        fhir_client.resources("Observation")
+        .limit(settings.MAX_RESOURCE_COUNT)
+        .include("Observation", "subject")
+        .fetch_raw()
+    )
+    for entry in [e for e in result.entry if e.search.mode == "match"]:
+        observation = entry.resource
+        included_ref = find_ref_in_bundle(result, observation.subject.reference)
```
Comment (Jasopaum) [inline tests/river/test_batch.py:125]:
```
ref_id = observation.subject.reference.split("/")[1]
assert (
    any(
        entry.resource.id == ref_id
        for entry in entries
        for entries in bundle.entry  # Or with the fors swapped, I don't remember
        if entry.search.mode == "include"
    ),
    f"missing referenced resource {observation.subject.reference}"
)
```

File: tests/river/test_cancel_batch.py (hunk 1 above)
Comment (simonvadee) [reply tests/river/test_cancel_batch.py:30]: I think a [200 is fine](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE): 
> A 200 (OK) status code if the action has been enacted and the response message includes a representation describing the status.

Comment (simonvadee) [reply tests/river/test_cancel_batch.py:30]: but you can make an issue if you think it's worth!

File: requirements/base.txt [hunk 5]
```diff
@@ -1,10 +1,9 @@
...
 elasticsearch==7.0.5
-fhirstore @ git+https://github.com/arkhn/pyfhirstore@0.5.0
-pymongo==3.9.0
 pytest==6.1.2
 python-dotenv==0.15.0
 requests==2.24.0
 setuptools-scm==4.1.2
 redis==3.5.3
 hiredis==1.1.0
+fhirpy==1.2.1
```
[the remark this answers is not in the document]
Comment (simonvadee) [reply requirements/base.txt:9]: sorry mom

[the remark this answers is not in the document]
Comment (tevariou) [reply requirements/base.txt:9]: about that.. How about switching from requirements.txt ? The official Python doc recommendation is to use Pipenv which uses Pipfile. I find it easier to add/remove dependencies. https://github.com/pypa/pipfile#the-concept

File: tests/river/conftest.py (hunk 3 above)
Comment (simonvadee) [reply tests/river/conftest.py:60]: no reason (fixed)

File: tests/river/test_batch.py (hunk 4 above)
Comment (simonvadee) [reply tests/river/test_batch.py]: done

File: requirements/base.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (simonvadee) [reply requirements/base.txt]: sorry mom

File: .vscode/settings.json [hunk 6]
```diff
@@ -1,4 +1,4 @@
 {
-    "python.pythonPath": ".venv/bin/python",
+    "python.pythonPath": "venv/bin/python",
```
[the remark this answers is not in the document]
Comment (simonvadee) [reply .vscode/settings.json:2]: ok: the convention is .venv

**Pull request merged**